src/utils.c

```diff
@@ -27,7 +27,7 @@ uint16_t avifHTONS(uint16_t s)
 uint16_t avifNTOHS(uint16_t s)
 {
     uint8_t data[2];
-    memcpy(&data, &s, sizeof(data));
+    memcpy(data, &s, sizeof(data));
 
     return (uint16_t)((data[1] << 0) | (data[0] << 8));
```

Comment:
Out of curiousity, why not just the following? Is it about speed? I do not see how it solves any endianness issue even after reading the blog article above.
```c++
const uint8_t* data = (const uint8_t*)&s;
return (uint16_t)((data[1] << 0) | (data[0] << 8));
```

Reply:
Yannis: I believe your code is equivalent and saves a copy.

Reply:
Usually when reinterpreting some memory (for example getting bit representation of float point value), `memcpy` is used to avoid violating strict aliasing rule (https://en.cppreference.com/w/c/language/object#Strict_aliasing). But as a special case `char`/`unsigned char` aliasing is allowed, so `memcpy` is not necessary here. Anyway, with proper optimization level set, the call to `memcpy` should be optimized away by most compilers.

Reply:
Thank you for the explanation.